server/src/main/java/org/kaazing/gateway/server/util/version/ClassPathParser.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2007-2014 Kaazing Corporation. All rights reserved.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.kaazing.gateway.server.util.version;
+
+import java.io.IOException;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+/**
+ * Class used for parsing the class path system property.
+ */
+public class ClassPathParser {
+
+    public ClassPathParser() {
+    }
+
+    /**
+     * Retrieves the class path entries.
+     * @return the class path entries
+     */
+    public String[] getClassPathEntries() {
+        return System.getProperty("java.class.path").split(System.getProperty("path.separator"));
```

Comment:
Rather than using classpath system property, I think loading of MANIFEST.MF resources using the classloader would be better.